src/manager/checkpoint.rs

```diff
@@ -243,6 +268,15 @@ async fn manage_subnet((child, parent): (Subnet, Subnet), stop_notify: Arc<Notif
     ))
 }
 
+/// Sleeps for some time if stop_notify is not fired. It returns true to flag that we should move to the
+/// next iteration of the loop, while false informs that the loop should return.
+async fn wait_next_iteration(stop_notify: &Arc<Notify>) -> Result<bool> {
+    select! {
+        _ = sleep(CHAIN_HEAD_REQUEST_PERIOD) => {Ok(true)}
+        _ = stop_notify.notified() => {Ok(false)}
```

Comment:
not super familiar with the `notified` api, I could be wrong, but just curious will there be missed signal. Consider the reloadable config had an update in config, then causing the `stop_notify` to invoke `notify_all`, but the checkpointing thread has not yet reached `wait_next_iteration`, and when reaching select, what happens to `stop_notify.notified()`, will it receive the signal or miss?

Reply:
@hmoniz, can you comment on this? TBH, I didn't write this logic and I would have to look deeper into it